### PR TITLE
Detect starting under incompatible Java version

### PR DESCRIPTION
**Pull request by vmax**
## What is the goal of this PR?

Warn user if an incompatible Java version is used to start up the server.

## What are the changes implemented in this PR?

Detect if server is started under incompatible version of Java (any before 11) and throw an error on startup

### CODE REVIEW

File: server/GraknServer.java [hunk 1]
```diff
@@ -127,6 +130,13 @@ private Path dataDir() {
         return command.dataDir();
     }
 
+    private void configureAndVerifyJavaVersion() {
+        int javaMajorVersion = Java.getMajorVersion();
+        if (javaMajorVersion < 11) {
```
Comment (lolski) [inline server/GraknServer.java:135]: I think there's no need for the variable:

```suggestion
        if (Java.getMajorVersion() < 11) {
```

File: common/exception/ErrorMessage.java [hunk 2]
```diff
@@ -69,6 +69,8 @@ public static void loadConstants() {
                 new Server(16, "The request with ID '%s' is a duplicate.");
         public static final Server ALREADY_RUNNING =
                 new Server(17, "Another instance of Grakn Core server is already running at this port: '%s'.");
+        public static final Server INCOMPATIBLE_JAVA_RUNTIME =
+                new Server(18, "Incompatible Java runtime version: '%s'.");
```
Comment (lolski) [inline common/exception/ErrorMessage.java:73]:
```suggestion
                new Server(18, "Incompatible Java runtime version: '%s'. Please use Java 11 or above.");
```

**Review by lolski — CHANGES_REQUESTED** [review]

File: server/GraknServer.java (hunk 1 above)
Comment (vmax) [reply server/GraknServer.java:135]: @lolski it's extracted into a variable to be later passed into an error message:

```
            throw GraknException.of(INCOMPATIBLE_JAVA_RUNTIME, javaMajorVersion);
```

File: server/GraknServer.java [hunk 3]
```diff
@@ -127,6 +130,16 @@ private Path dataDir() {
         return command.dataDir();
     }
 
+    private void configureAndVerifyJavaVersion() {
+        int javaMajorVersion = Java.getMajorVersion();
+        if (javaMajorVersion == Java.UNKNOWN_VERSION) {
+            LOG.warn("Could not detect Java version from version string '{}'", System.getProperty("java.version"));
```
Comment (lolski) [inline server/GraknServer.java:136]: `Could not detect Java version from version string '{}'. Will start Grakn Core anyway.`

Comment (vmax) [reply server/GraknServer.java]: @lolski addressed in https://github.com/graknlabs/grakn/pull/6251/commits/e43116bd6946d31ea988551711a5435d37906db6

**Review by lolski — APPROVED** [review]

File: server/GraknServer.java [hunk 4]
```diff
@@ -127,6 +130,16 @@ private Path dataDir() {
         return command.dataDir();
     }
 
+    private void configureAndVerifyJavaVersion() {
+        int javaMajorVersion = Java.getMajorVersion();
+        if (javaMajorVersion == Java.UNKNOWN_VERSION) {
+            LOG.warn("Could not detect Java version from version string '{}'. Will start Grakn Core anyway.", System.getProperty("java.version"));
```
Comment (lolski) [inline server/GraknServer.java:136]: Let's actually say `Will start Grakn Core Server anyway`

File: server/GraknServer.java [hunk 5]
```diff
@@ -127,6 +130,16 @@ private Path dataDir() {
         return command.dataDir();
     }
 
+    private void configureAndVerifyJavaVersion() {
+        int javaMajorVersion = Java.getMajorVersion();
```
Comment (lolski) [inline server/GraknServer.java:134]: Let's rename the variable to `majorVersion`

File: server/GraknServer.java (hunk 4 above)
Comment (vmax) [reply server/GraknServer.java]: @lolski addressed in https://github.com/graknlabs/grakn/pull/6251/commits/6958bb9122458f7786ca06b66786c338b46b1240

File: server/GraknServer.java (hunk 5 above)
Comment (vmax) [reply server/GraknServer.java]: @lolski addressed in https://github.com/graknlabs/grakn/pull/6251/commits/6958bb9122458f7786ca06b66786c338b46b1240

**Review by lolski — APPROVED** [review]

**Pull request merged**